modules/dnn/include/opencv2/dnn/all_layers.hpp

```diff
@@ -473,6 +473,7 @@ CV__DNN_EXPERIMENTAL_NS_BEGIN
         float epsilon;
 
         virtual void getScaleShift(Mat& scale, Mat& shift) const = 0;
+        virtual void getScaleShift(UMat& scale, UMat& shift) const = 0;
```

Comment:
This call is not necessary. It is used during preparation stage, so there is no problem to use Mat instead.

Also
```
         umat_weight = weights_.getUMat(ACCESS_READ);
         umat_bias = bias_.getUMat(ACCESS_READ);
```

in batch_norm_layer.cpp should not be initialized unconditionally.

On systems with OpenCL without "zero-copy" support memory consumption would be too large (especially when OpenCL is not used).
It is better to initialize this due the first call of the "forward()" processing.

Reply:
it makes sense to me.  code is updated accordingly